clients/include/testing_matmul.hpp

```diff
@@ -330,42 +317,89 @@ void check(hipStream_t                          stream,
             }
             if(arg.gradient && arg.bias_vector)
             {
-                if(arg.d_type != arg.scale_type && arg.bias_type == arg.scale_type)
-                {
-                    double norm_error
-                        = std::abs(norm_check_general<Talpha>('F',
-                                                              M[gemmIdx],
-                                                              1,
-                                                              M[gemmIdx],
-                                                              M[gemmIdx],
-                                                              *(hBias_gold_C[gemmIdx]),
-                                                              *(hBias_C[gemmIdx]),
-                                                              num_batches[gemmIdx]));
-                    hipblaslt_error += norm_error;
-                    if(arg.norm_check_assert)
-                        CHECK_SUCCESS(norm_check<Talpha>(norm_error));
-                }
-                else
-                {
-                    double norm_error = std::abs(norm_check_general<To>('F',
+                double norm_error = std::abs(norm_check_general<Tbias>('F',
                                                                         M[gemmIdx],
                                                                         1,
                                                                         M[gemmIdx],
                                                                         M[gemmIdx],
                                                                         *(hBias_gold[gemmIdx]),
                                                                         *(hBias[gemmIdx]),
                                                                         num_batches[gemmIdx]));
-                    hipblaslt_error += norm_error;
-                    if(arg.norm_check_assert)
-                        CHECK_SUCCESS(norm_check<To>(norm_error));
-                }
+                hipblaslt_error += norm_error;
+                if(arg.norm_check_assert)
+                    CHECK_SUCCESS(norm_check<Tbias>(norm_error));
             }
         }
     }
 }
 
+// A function to determing the default bias_type
+hipDataType derive_unset_bias_type(const Arguments& arg)
+{
+    static const std::set<hipDataType> supported_bias_types
+        = {HIP_R_32F, HIP_R_64F, HIP_R_16F, HIP_R_16BF, HIP_R_32I};
+
+    hipDataType real_bias_type = arg.bias_type;
+    
+    // when bias type is unset.
+    if(arg.bias_type == HIPBLASLT_DATATYPE_INVALID)
+    {
+        if(arg.compute_type == HIPBLAS_COMPUTE_32I || arg.compute_type == HIPBLAS_COMPUTE_32F_FAST_TF32)
```

Comment:
This function follows the logic of RocblasltContractionProblem() in tensile_host.hpp.